buildSrc/src/main/groovy/com/google/firebase/gradle/GenerateMeasurementsTask.groovy

```
@@ -0,0 +1,182 @@
+ * The report path is where the output should be stored. Additionally, a project property, {@code
+ * pull_request} is also required. This value will be placed in the report.
+ */
+public class GenerateMeasurementsTask extends DefaultTask {
+
+    /**
+     * The file storing the SDK map.
+     *
+     * <p>This may be any type recognized by Gradle as a file. The format of the file's contents is
+     * headerless CSVwith a colon as a delimiter: projectName_buildVariant:sdkId. The first column
+     * contains both the project name and build variant separated by an hyphen, {@code
+     * firestore-aggressive}, for example.
+     */
+    @InputFile
+    File sdkMap
+
+    /**
+     * The file for storing the report.
+     *
+     * <p>This may be any type recognized by Gradle as a file. The contents, if any, will be
+     * overwritten by the new report.
+     */
+    @OutputFile
+    File report
+
+    @Override
+    Task configure(Closure closure) {
+        project.android.variantFilter {
+            if (it.buildType.name != "aggressive") {
+                it.ignore = true;
+            }
+        }
+
+        outputs.upToDateWhen { false }
+        dependsOn "assemble"
+        return super.configure(closure)
+    }
+
+    @TaskAction
+    def generate() {
+        def sdks = createSdkMap()
+        def sizes = calculateSizes(sdks, project.android.applicationVariants)
+        def report = createReport(sizes)
+
+        project.file(this.report).withWriter {
+            it.write(report)
+        }
+    }
+
+    private def calculateSizes(sdks, variants) {
+      def sizes = [:]
+
+      variants.each { variant ->
+        def name = "${variant.flavorName}-${variant.buildType.name}"
+        def apks = variant.outputs.findAll { it.outputFile.name.endsWith(".apk") }
+        if (apks.size() > 1) {
+            throw new IllegalStateException("${name} produced more than one APK")
+        }
```

Do we fail silently if < 1?

---

We skip that variant if there's no APK. I'll add a comment better clarifying the behavior.

I don't think there's a reason for the uploader to fail if a previous build step failed to produce an APK.